src/DynamoCoreWpf/Views/HomePage/HomePage.xaml.cs

```diff
@@ -471,14 +478,51 @@ internal void ApplicationLoaded()
 
         #endregion
 
+        #region Helper Functions
+        /// <summary>
+        /// Create unique subfolder to allocate webView2 resources
+        /// </summary>
+        /// <param name="baseDir"></param>
+        /// <returns></returns>
+        private DirectoryInfo CreateUniqueUserDataFolder(string baseDir)
+        {
+            Directory.CreateDirectory(baseDir);
+
+            var uniqueSubfolderName = Guid.NewGuid().ToString();
+            var uniquePath = Path.Combine(baseDir, uniqueSubfolderName);
+            Directory.CreateDirectory(uniquePath);
+
+            return new DirectoryInfo(uniquePath);
+        }
+        #endregion
+
         #region Dispose
         public void Dispose()
         {
             DataContextChanged -= OnDataContextChanged;
-            if(startPage != null) startPage.DynamoViewModel.PropertyChanged -= DynamoViewModel_PropertyChanged;
 
-            this.dynWebView.CoreWebView2.NewWindowRequested -= CoreWebView2_NewWindowRequested;
+            if(startPage != null) startPage.DynamoViewModel.PropertyChanged -= DynamoViewModel_PropertyChanged;
+            if (this.dynWebView != null)
+            {
+                var userDataFolder = this.dynWebView.CoreWebView2.Environment.UserDataFolder;
+                if (userDataFolder != null && Directory.Exists(userDataFolder))
+                {
+                    // Try shutting down the browser with Dispose, and wait for process to exit
+                    try
+                    {
+                        var webViewProcessId = Convert.ToInt32(this.dynWebView.CoreWebView2.BrowserProcessId);
+                        var webViewProcess = Process.GetProcessById(webViewProcessId);
+
+                        this.dynWebView.Dispose();
+                        webViewProcess.WaitForExit(3000);
```

Comment:
you might be able to use this event `dynWebView.CoreWebView2.Environment.BrowserProcessExited`

Reply:
Oh, great, I can definitely do that. Thank you for the suggestion.

Reply:
probably have to clean it up too..